### PR TITLE
enhancement(observability): API host metrics

### LINKED ISSUE #4649: Add host metrics to the GraphQL API
Consume the `host_metrics` source in the API, outside of source configuration.

This should be a 'hot' source - i.e. only active when requested, and relatively light; metrics should be collected only if the pertinent fields are queried.

The use-case is memory/CPU gauges in `vector top`.

### PR DESCRIPTION
**Pull request by leebenson**
Exposes host metrics via the GraphQL API:

Screenshot 2020-10-20 at 11 52 35

Closes #4649.

## Demo

https://www.loom.com/share/bb3c5274cd0440c88217743a6a6eb1f5?from_recorder=1

## Example query

```gql
{
  hostMetrics {
    memory {
      totalBytes
      freeBytes
      availableBytes
      activeBytes
      buffersBytes
      cachedBytes
      sharedBytes
      usedBytes
      inactiveBytes
      wiredBytes
    }
    swap {
      freeBytes
      totalBytes
      usedBytes
      swappedInBytesTotal
      swappedOutBytesTotal
    }
    cpu {
      cpuSecondsTotal
    }
    loadAverage {
      load1
      load5
    }
    network {
      receiveBytesTotal
      receiveErrsTotal
      receivePacketsTotal
      transmitBytesTotal
      transmitErrsTotal
      transmitPacketsDropTotal
      transmitPacketsTotal
    }
    filesystem {
      freeBytes
      totalBytes
      usedBytes
    }
    disk {
      readBytesTotal
      readsCompletedTotal
      writtenBytesTotal
      writesCompletedTotal
    }
  }
}
```

Returning:

```json
{
  "data": {
    "hostMetrics": {
      "cpu": {
        "cpuSecondsTotal": 424616.36
      },
      "disk": {
        "readBytesTotal": 1485091422208,
        "readsCompletedTotal": 42045267,
        "writesCompletedTotal": 19533505,
        "writtenBytesTotal": 1322889637888
      },
      "filesystem": {
        "freeBytes": 470843863040,
        "totalBytes": 1000240963584,
        "usedBytes": 11245162496
      },
      "loadAverage": {
        "load1": 2.58154296875,
        "load5": 2.8662109375
      },
      "memory": {
        "activeBytes": 11843481600,
        "availableBytes": 15732486144,
        "buffersBytes": null,
        "cachedBytes": null,
        "freeBytes": 1790156800,
        "inactiveBytes": 10031222784,
        "sharedBytes": null,
        "totalBytes": 34359738368,
        "usedBytes": null,
        "wiredBytes": 4946178048
      },
      "network": {
        "receiveBytesTotal": 786269184,
        "receiveErrsTotal": 0,
        "receivePacketsTotal": 532194,
        "transmitBytesTotal": 786269184,
        "transmitErrsTotal": 0,
        "transmitPacketsDropTotal": null,
        "transmitPacketsTotal": null
      },
      "swap": {
        "freeBytes": 935854080,
        "swappedInBytesTotal": 1278002446336,
        "swappedOutBytesTotal": 1188851712,
        "totalBytes": 9663676416,
        "usedBytes": 8727822336
      }
    }
  }
}
```

### CODE REVIEW

File: src/api/schema/metrics/mod.rs [hunk 1]
```diff
@@ -1,4 +1,9 @@
-use crate::event::{Event, Metric, MetricValue};
+mod bytes_processed;
```
Comment (leebenson) [inline src/api/schema/metrics/mod.rs:1]: Moved existing metrics into a `metrics` sub-mod, to accommodate this growing area of the API schema

File: src/sources/host_metrics.rs [hunk 2]
```diff
@@ -463,7 +463,7 @@ impl HostMetricsConfig {
                                 self.counter(
                                     "network_receive_packets_total",
                                     timestamp,
-                                    counter.drop_recv() as f64,
+                                    counter.packets_recv() as f64,
```
Comment (leebenson) [inline src/sources/host_metrics.rs:466]: Noticed this was pulling the wrong stat

**Comment by leebenson** [comment]
@jamtur01, tagging you on this-- resolved a few conflicts from #4647, so just a sanity check that the schema matches conventions.

**Review by jamtur01 — APPROVED** [review]
LGTM

File: src/sources/host_metrics.rs [hunk 3]
```diff
@@ -101,7 +101,7 @@ impl Namespace {
 }
 
 #[derive(Clone, Debug, Default, Deserialize, Serialize)]
-struct HostMetricsConfig {
+pub struct HostMetricsConfig {
```
Comment (Hoverbear) [inline src/sources/host_metrics.rs:104]: I imagine these would be better suited as `pub(crate)`, but that's not a blocking comment :)

**Review by Hoverbear — APPROVED** [review]
This looks good, @leebenson ! I didn't much review the json changes since they are generated.

**Pull request merged**